File: backend/src/YouG.Domain/Entities/EventLocationOption.cs
using YouG.Domain.Common;

namespace YouG.Domain.Entities;

public class EventLocationOption : Entity
{
    public required Guid EventId { get; set; }
    public required string Name { get; set; }
    public string? Address { get; set; }
    public required double Latitude { get; set; }
    public required double Longitude { get; set; }
    public required Guid ProposedByUserId { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
}

File: mobile/lib/features/events/domain/entities/event_location_option.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'event_location_option.freezed.dart';

@freezed
abstract class EventLocationOption with _$EventLocationOption {
  const factory EventLocationOption({
    required String id,
    required String name,
    String? address,
    required double latitude,
    required double longitude,
    required String proposedByUserId,
    required int voteCount,
    required bool hasCurrentUserVoted,
  }) = _EventLocationOption;
}
